src/ape_ethereum/ecosystem.py

```diff
@@ -101,23 +103,24 @@ class NetworkConfig(PluginConfig):
     class Config:
         smart_union = True
 
-    @validator("gas_limit", pre=True, allow_reuse=True)
-    def validate_gas_limit(cls, value: GasLimit) -> GasLimit:
-        if isinstance(value, str):
-            if value.lower() in ("auto", "max"):
-                return value.lower()
-
-            # Value could be an integer string
-            if value.isdigit():
-                return int(value)
-            # Enforce "0x" prefix on base 16 integer strings
-            elif value.lower().startswith("0x"):
-                return int(value, 16)
-            else:
-                raise ValueError("Invalid gas_limit, must be 'auto', 'max', or a number")
+    @validator("gas_limit", pre=True)
```

Comment:
I think `allow_reuse` is necessary so that subclasses which use this class as a base inherit this validator also

Reply:
ya good catch, a lot of ecosystems subclass `Ethereum`.

Reply:
oh but this is only the config class, not sure if that gets subclassed ever, but it doesn't hurt I suppose

Reply:
I put it in, just in case